addons/binding/org.openhab.binding.exec/src/main/java/org/openhab/binding/exec/handler/ExecHandler.java

```diff
@@ -53,191 +58,326 @@
-                            new Object[] { commandLine.toString(), e.getMessage() });
-                }
+                    boolean exitVal = false;
+                    try {
+                        exitVal = proc.waitFor(timeOut, TimeUnit.MILLISECONDS);
+                    } catch (InterruptedException e) {
+                        logger.error("An exception occurred while waiting for the process ('{}') to finish : '{}'",
+                                new Object[] { commandLine.toString(), e.getMessage() });
+                    }
 
-                if (!exitVal) {
-                    logger.warn("Forcibly termininating the process ('{}') after a timeout of {} ms",
-                            new Object[] { commandLine.toString(), timeOut });
-                    proc.destroyForcibly();
-                }
+                    if (!exitVal) {
+                        logger.warn("Forcibly termininating the process ('{}') after a timeout of {} ms",
+                                new Object[] { commandLine.toString(), timeOut });
+                        proc.destroyForcibly();
+                    }
 
-                updateState(RUN, OnOffType.OFF);
-                updateState(EXIT, new DecimalType(proc.exitValue()));
+                    updateState(RUN, OnOffType.OFF);
+                    updateState(EXIT, new DecimalType(proc.exitValue()));
 
-                outputBuilder.append(errorBuilder.toString());
+                    outputBuilder.append(errorBuilder.toString());
 
-                outputBuilder.append(errorBuilder.toString());
+                    String transformedResponse = StringUtils.chomp(outputBuilder.toString());
+                    String transformation = (String) getConfig().get(TRANSFORM);
 
-                String transformedResponse = StringUtils.chomp(outputBuilder.toString());
-                String transformation = (String) getConfig().get(TRANSFORM);
+                    if (transformation != null && transformation.length() > 0) {
+                        transformedResponse = transformString(transformedResponse, transformation);
+                    }
 
-                if (transformation != null && transformation.length() > 0) {
-                    transformedResponse = transformResponse(transformedResponse, transformation);
-                }
+                    updateState(OUTPUT, new StringType(transformedResponse));
 
-                updateState(OUTPUT, new StringType(transformedResponse));
+                    DateTimeType stampType = new DateTimeType(Calendar.getInstance());
+                    updateState(LAST_EXECUTION, stampType);
 
-                DateTimeType stampType = new DateTimeType(Calendar.getInstance());
-                updateState(LAST_EXECUTION, stampType);
+                }
 
+            } catch (Exception e) {
+                logger.error("An exception occurred while executing a command : '{}'", e.getMessage(), e);
```

Comment:
Should you mention which command? Should you catch all exceptions, what kind do you expect? Should it be error level?

Note that the finally will even be triggered if we do not catch the exception.